PER-40 Masterdata script and testdata

nihallakshmankonchady opened the pull request.

Commit:
PER-40 Masterdata script and testdata

Signed-off-by: Nihal LK <nihal.lk@technoforte.co.in>

Sanjayss04 commented:
Always try to give some description to the PR, so it is easy to check on the updated points.

Sanjayss04 commented on commons/masterdata/scripts/Masterdata_Test_Script.jmx:
```diff
@@ -53,41 +53,64 @@
       </CSVDataSet>
       <hashTree/>
-      <ThreadGroup guiclass="ThreadGroupGui" testclass="ThreadGroup" testname="Get Templates (Preparation)" enabled="true">
+      <Arguments guiclass="ArgumentsPanel" testclass="Arguments" testname="User Defined Variables - Auth token generation" enabled="true">
```
Why 2 user defined variables are added in the test plan ? Can't we have only one ?

nihallakshmankonchady replied:
@Sanjayss04 .. Made it to one user-defined config.

Sanjayss04 commented on commons/masterdata/scripts/Masterdata_Test_Script.jmx:
```diff
@@ -53,41 +53,64 @@
+        </collectionProp>
+      </Arguments>
+      <hashTree/>
+      <ThreadGroup guiclass="ThreadGroupGui" testclass="ThreadGroup" testname="Auth Token Generation (Setup)" enabled="true">
```
Auth Token Generation (Setup) is added in the Test Script itself. Are we not using Helper script for auth token generation?

nihallakshmankonchady replied:
@Sanjayss04 ... Auth Token Generation (Setup) has been removed from Test Script.

Sanjayss04 commented on commons/masterdata/scripts/Masterdata_Test_Script.jmx:
```diff
@@ -53,41 +53,64 @@
-        </CSVDataSet>
-        <hashTree/>
-        <HTTPSamplerProxy guiclass="HttpTestSampleGui" testclass="HTTPSamplerProxy" testname="Create Templates" enabled="true">
+        <HTTPSamplerProxy guiclass="HttpTestSampleGui" testclass="HTTPSamplerProxy" testname="Authentication API" enabled="true">
```
As per the Performance developer guide it is mentioned that every sampler must have 2 types of assertion checks available, text and response code assertion.
I could not see response code assertion in any of the samplers. Please add it at all places.

nihallakshmankonchady replied:
@Sanjayss04 ... Added the assertion for 200 response code.

Sanjayss04 commented on commons/masterdata/scripts/Masterdata_Test_Script.jmx:
```diff
@@ -314,7 +341,7 @@
           <stringProp name="shareMode">shareMode.all</stringProp>
         </CSVDataSet>
         <hashTree/>
-        <HTTPSamplerProxy guiclass="HttpTestSampleGui" testclass="HTTPSamplerProxy" testname="Get Templates By Lang &amp; Temp Code API Execution" enabled="true">
```
Please check for the variable name under Get Templates By Lang & Temp Code (Preparation) thread group. Don't use "lang&TempTypeCode" as it contains a special character in between. 
Replace it with "langTempTypeCode"

nihallakshmankonchady replied:
Renamed with langTempTypeCode.

nihallakshmankonchady commented:
Updated the Masterdata Helper and Test Script with some minor modifications.

- Updated the CSV file with correct values. The endpoints are running fine now.
- Added response code assertion.
- Corrected the naming convention of test data files which are used in the Test Script.

nihallakshmankonchady commented:
@Sanjayss04 .. i have added the description for the PR.

Commit:
Updated the scripts with review comments

Signed-off-by: Nihal LK <nihal.lk@technoforte.co.in>
